components/brave_wallet_ui/common/constants/solana.ts

```diff
@@ -44,3 +19,118 @@ export const SolanaTransactionTypes = [
   BraveWallet.TransactionType.SolanaDappSignAndSendTransaction,
   BraveWallet.TransactionType.SolanaSwap
 ]
+
+export const SOL_ACCOUNT_INSTRUCTION_PARAM_NAMES = {
+  account: 'account',
+  allocated_account: 'allocated_account',
+  assigned_account: 'assigned_account',
+  authority: 'authority',
+  base_account: 'base_account',
+  burned_account: 'burned_account',
+  changed_mint_or_account: 'changed_mint_or_account',
+  closed_account: 'closed_account',
+  created_account: 'created_account',
+  delegate: 'delegate',
+  from_account: 'from_account',
+  frozen_account: 'frozen_account',
+  initialized_account: 'initialized_account',
+  initialized_mint: 'initialized_mint',
+  initialized_multisig_account: 'initialized_multisig_account',
+  mint: 'mint',
+  native_token_account: 'native_token_account',
+  new_account: 'new_account',
+  nonce_account: 'nonce_account',
+  nonce_authority: 'nonce_authority',
+  owner_delegate: 'owner_delegate',
+  owner: 'owner',
+  recentblockhashes_sysvar: 'recentblockhashes_sysvar',
+  rent_sysvar: 'rent_sysvar',
+  signers: 'signers',
+  to_account: 'to_account',
+  token_mint: 'token_mint'
+} as const
+
+export const SOL_INSTRUCTION_PARAM_NAMES = {
+  amount: 'amount',
+  authority_type: 'authority_type',
+  authority: 'authority',
+  base: 'base',
+  decimals: 'decimals',
+  freeze_authority: 'freeze_authority',
+  from_owner_program: 'from_owner_program',
+  from_seed: 'from_seed',
+  lamports: 'lamports',
+  mint_authority: 'mint_authority',
+  new_authority: 'new_authority',
+  num_of_signers: 'num_of_signers',
+  owner_program: 'owner_program',
+  owner: 'owner',
+  seed: 'seed',
+  space: 'space'
+} as const
+
```

Comment:
Is this something we can expose via mojo?

Reply:
If possible that would be nice, we have similar on iOS (though only `lamports`, `amount`, and `decimals` are used currently):
https://github.com/brave/brave-ios/blob/development/Sources/BraveWallet/Extensions/SolanaInstruction%2BExtensions.swift#L145-L163

Reply:
I couldn't get Mojom to expose a constant array or a struct with constants as properties.
Here I exposed the list as a constant string, which can be parsed into an array by the frontends
I added notes above the types to explain where the values came from

https://github.com/brave/brave-core/pull/17128/files#diff-9f2cf11d57c8efee5461b92cdb38fa327b70c25fd8d009f7692049e8e88944a8R906-R915

Reply:
@yrliou  & @StephenHeaps , how do you feel about that ☝️ ? Is there a better way to share this info as a constant?

Reply:
Program Id constants are fine, currently we only display either `System Program` or `Token Program` (otherwise we display `Unknown Instruction` w/ accounts & data) on iOS but we should be able to add these new values similarly and add localizations:
https://github.com/brave/brave-ios/blob/development/Sources/BraveWallet/Extensions/SolanaInstruction%2BExtensions.swift#L10-L39

I'm not sure we would make use of `kSolanaInstructionAddressTypeParamNames` on iOS (maybe I'm misunderstanding the usage here). If they were constants similar to the program ids rather than a comma separated string we could make better use of some of these, but given we only use `to_account`, `from_account`, `lamports`, `amount` and `decimals` are used directly for us I'm not sure this is a big issue.
ex. 
if we had
`const string kSolanaInstructionParamNameToAccount = "to_account";`
we could make use of these like so:
```
  /// Returns the `to_account` pubkey for the instruction if available
  var toPubkey: String? {
    guard let index = decodedData?.accountParams.firstIndex(where: { $0.name == BraveWallet.SolanaInstructionParamNameToAccount }) else { return nil }
    return accountMetas[safe: index]?.pubkey
  }
```

Reply:
Just for future reference, this comment has been resolved by a backend PR https://github.com/brave/brave-core/pull/17267.